api/v1beta2/awscluster_webhook.go

```diff
@@ -163,6 +164,31 @@ func (r *AWSCluster) ValidateUpdate(old runtime.Object) error {
 		)
 	}
 
+	if gcTasksAnotationValue := r.GetAnnotations()[ExternalResourceGCTasksAnnotation]; gcTasksAnotationValue != "" {
```

Comment:
Just being careful here. What if the annotation does exist but it _IS_ empty?

We should probably rather say something like `v, ok := r.GetAnnotations()[ExternalResourceGCTasksAnnotation]; ok` ?

Reply:
if the annotation exists and its value is empty, then there is nothing to validate and we don't enter this section